cli/cmd/compose/up.go

```diff
@@ -144,7 +149,25 @@ func runCreateStart(ctx context.Context, opts upOptions, services []string) erro
 		return nil
 	}
 
-	err = c.ComposeService().Start(ctx, project, formatter.NewLogConsumer(ctx, os.Stdout))
+	ctx, cancel := context.WithCancel(ctx)
+	listener := make(chan compose.Event)
+	go func() {
+		var aborting bool
+		for {
+			<-listener
+			if opts.cascadeStop && !aborting {
```

Comment:
We can avoid the whole goroutine and context change if `!opts.cascadeStop`, and then maybe remove the forever loop in the goroutine ?